content/blog/debugging-nodejs-programs-using-chrome/index.md

```diff
@@ -0,0 +1,308 @@
+
+![The "step outside" button being highlighted with a circle with the line 12 console log being paused](./step-outside.png)
+
+> While the example uses a callback in `map`, both of these "step into" and "step out of" also work on functions that are called. For example, assume the code was written as the following:
+>
+> ```javascript
+> const getEmployeeAges = partialList => {
+>   const ageArray = [];
+>   for (employee of partialList) {
+>     ageArray.push(employee.employee_age);
+>   }
+>   return ageArray;
+> };
+> 
+> app.get('/', (req, res) => {
+>   request('http://www.mocky.io/v2/5e1a9abe3100004e004f316b', (error, response, body) => {
+>     const responseList = JSON.parse(body);
+>     const partialList = responseList.slice(0, 20);
+>     const employeeAges = getEmployeeAges(partialList);
+>     console.log(employeeAges);
+>   });
+> });
+> ```
+>
+> You would still be able to "step into" `getEmployeeAges` and, once inside, "step outside" again in the same manor of the `map` function, as shown prior.
+
+# Saving Files {#editing-files-in-chrome}
+
+One more feature I'd like to touch on with the debugger before closing things out is the ability to edit the source files directly from the debugger. Using this feature, it can make the Chrome debugger a form of lite IDE, which may improve your workflow. So, let's revert our code to [the place it was at before we applied the fix we needed](#example-code) and go from there.
+
+![The screenshot of the debugger with the original code](./initial-debugger.png)
+
+Once this window is open, you're able to tab into or use your cursor to select within the text container that holds your code. Once inside, it should work just like a `textarea`, which _allows you to change code as you might expect from any other code editor_. Changing line `10` to `return employee.employee_age` instead of `return employee.employeeAge` will show an astererisk (`*`) to let you know your changes are not persisting. _Running your code in this state will not reflect the changes made to the code content on the screen_, which may have unintended effects.
+
+![Showing the asterisk once changes made but not saved](./edited-but-not-saved.png)
+
+In order to make your changes persist, you'll need to press `Ctrl + S` or `Command + S` in order to save the file (much like a Word document). Doing so will bring up a yellow triangle instead of an asterisk indicating _your changes are not saved to the source file but your changes will now take effect_. Re-running the `localhost:3000` route will now correct the behavior you're wanting, but if you open `app.js` in a program like Visual Studio Code, it will show the older broken code.
```

Comment:
```suggestion
In order to make your changes persist, you'll need to press `Ctrl + S` or `Command + S` in order to save the file (much like a Word document). Doing so will bring up a yellow triangle instead of an asterisk indicating _your changes are not saved to the source file but your changes will now take effect_. Re-running the `localhost:3000` route will now correct the behavior you want, but if you open `app.js` in a program like Visual Studio Code, it will show the older broken code.
```